nibabel/tests/test_image_types.py

```diff
@@ -0,0 +1,186 @@
+# emacs: -*- mode: python-mode; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the NiBabel package for the
+#   copyright and license terms.
+#
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+''' Tests for is_image / may_contain_header functions '''
+from __future__ import division, print_function, absolute_import
+
+import copy
+from os.path import dirname, basename, join as pjoin
+
+import numpy as np
+
+from .. import (Nifti1Image, Nifti1Header, Nifti1Pair,
+                Nifti2Image, Nifti2Header, Nifti2Pair,
+                AnalyzeImage, AnalyzeHeader,
+                Minc1Image, Minc2Image,
+                Spm2AnalyzeImage, Spm99AnalyzeImage,
+                MGHImage, all_image_classes)
+
+from nose.tools import assert_true, assert_equal, assert_raises
+
+DATA_PATH = pjoin(dirname(__file__), 'data')
+
+
+def test_analyze_detection():
+    # Test detection of Analyze, Nifti1 and Nifti2
+    # Algorithm is as described in loadsave:which_analyze_type
```

Comment:
I think this function was added in hopes of getting test coverage back up, but I don't think it explores paths that the other tests don't, so I'm for dropping it. That said, I don't want to abuse write access to @bcipolli's repo to remove code he added before he has a chance to defend it, so I'll leave that discussion there.